aten/src/ATen/native/Sorting.cpp

```diff
@@ -216,6 +217,10 @@ std::tuple<Tensor&, Tensor&> kthvalue_out_impl_cpu(
   // FIXME: This seems bogus, I only do this because it was the old behaviour.
   //        The reductions are fine, as long as the axis being reduced along
   //        isn't of 0 elements (and the output has elements).
+  auto lap = at::get_overlap_status(self, values);
```

Comment:
Can you put your code after the below '0 element check'? The above comments is related to '0 element check' IMO, we'd better keep them together.

Reply:
Done!